File: scripts/run-migrations.mjs
#!/usr/bin/env node
/**
 * Roda todas as migrations em ordem. Idempotente (CREATE TABLE IF NOT EXISTS).
 * Uso: npm run db:migrate
 *
 * Usa `postgres` lib (não @neondatabase/serverless): o driver Neon serverless
 * quebra em multi-statement scripts no novo host format .c-N. (gotcha conhecido).
 */
import postgres from "postgres";
import { readdirSync, readFileSync } from "fs";
import { join, dirname } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const MIGRATIONS_DIR = join(__dirname, "..", "migrations");

const url = process.env.DATABASE_URL;
if (!url) {
  console.error("DATABASE_URL not set. Coloque em .env.local e rode com `npm run db:migrate`.");
  process.exit(1);
}

const sql = postgres(url, { prepare: false, max: 1 });

const files = readdirSync(MIGRATIONS_DIR).filter((f) => f.endsWith(".sql")).sort();
console.log(`Encontradas ${files.length} migrations:`);
for (const f of files) console.log(`  - ${f}`);
console.log();

for (const f of files) {
  process.stdout.write(`▶ ${f} ... `);
  const sqlText = readFileSync(join(MIGRATIONS_DIR, f), "utf-8");
  try {
    await sql.unsafe(sqlText);
    console.log("ok");
  } catch (e) {
    console.error("FALHOU");
    console.error(e);
    await sql.end();
    process.exit(1);
  }
}

await sql.end();
console.log("\nMigrations concluídas.");

File: scripts/run-migrations.py
#!/usr/bin/env python3
"""Roda migrations SQL em ordem usando psycopg2 (mais robusto que o driver
serverless do Neon para CREATE TABLE/INDEX em massa)."""
import os, sys, glob

try:
    import psycopg2
except ImportError:
    print("python3 -m pip install --user psycopg2-binary")
    sys.exit(1)

BASE = os.path.dirname(os.path.abspath(__file__))
MIGRATIONS = sorted(glob.glob(os.path.join(BASE, "..", "migrations", "*.sql")))

url = os.environ.get("DATABASE_URL")
if not url:
    envpath = os.path.join(BASE, "..", ".env.local")
    if os.path.exists(envpath):
        with open(envpath) as f:
            for line in f:
                if line.startswith("DATABASE_URL="):
                    url = line.split("=", 1)[1].strip().strip('"').strip("'")
                    break
if not url:
    print("DATABASE_URL not set")
    sys.exit(1)

conn = psycopg2.connect(url)
conn.autocommit = True

for path in MIGRATIONS:
    name = os.path.basename(path)
    print(f"▶ {name} ... ", end="", flush=True)
    with open(path) as f:
        sql = f.read()
    with conn.cursor() as cur:
        try:
            cur.execute(sql)
            print("ok")
        except Exception as e:
            print(f"FALHOU: {e}")
            sys.exit(1)

conn.close()
print("\nMigrations concluídas.")
